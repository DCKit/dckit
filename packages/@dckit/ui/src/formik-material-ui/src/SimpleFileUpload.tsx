import * as React from 'react';
import { useField, useFormikContext } from 'formik';
import FormControl from '@material-ui/core/FormControl';
import InputLabel, { InputLabelProps } from '@material-ui/core/InputLabel';
import Input, { InputProps } from '@material-ui/core/Input';
import FormHelperText from '@material-ui/core/FormHelperText';

export interface SimpleFileUploadProps {
  name: string;
  label: string;
  disabled?: boolean;
  InputProps?: Omit<InputProps, 'name' | 'type' | 'onChange'>;
  InputLabelProps?: InputLabelProps;
}

export const SimpleFileUpload = ({
  name,
  label,
  disabled = false,
  InputProps: inputProps,
  InputLabelProps: inputLabelProps,
}: SimpleFileUploadProps) => {
  const { isSubmitting } = useFormikContext();
  const [field, meta, helpers] = useField(name);
  const error = meta.touched && meta.error;

  return (
    <div>
      <FormControl>
        {label && (
          <InputLabel shrink error={!!error} {...inputLabelProps}>
            {label}
          </InputLabel>
        )}
        <Input
          error={!!error}
          inputProps={{
            type: 'file',
            disabled: disabled || isSubmitting,
            name: field.name,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            onChange: (event: React.ChangeEvent<any>) => {
              const file = event.currentTarget.files[0];
              helpers.setValue(file);
            },
          }}
          {...inputProps}
        />
        {error && <FormHelperText error>{error}</FormHelperText>}
      </FormControl>
    </div>
  );
};
