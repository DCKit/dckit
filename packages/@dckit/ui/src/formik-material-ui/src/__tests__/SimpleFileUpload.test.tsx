import * as React from 'react';
import { Formik, Form } from 'formik';
import renderer from 'react-test-renderer';

import { SimpleFileUpload } from '../SimpleFileUpload';

test('SimpleFileUpload Renders Correctly', () => {
  const component = renderer.create(
    <Formik onSubmit={() => {}} initialValues={{ test: '' }}>
      <Form>
        <SimpleFileUpload name="test" label="Select" />
      </Form>
    </Formik>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
