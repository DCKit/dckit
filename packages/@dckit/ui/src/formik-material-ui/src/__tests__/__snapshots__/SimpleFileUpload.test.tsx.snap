// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleFileUpload Renders Correctly 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div>
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink"
        data-shrink={true}
      >
        Select
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="test"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="file"
        />
      </div>
    </div>
  </div>
</form>
`;
